spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultSchemaResolver.java

```diff
@@ -103,7 +116,64 @@ private Schema<?> getCustomSchemaOrMaybeDefault(Class<?> messageClass, boolean r
 		return this.customSchemaMappings.getOrDefault(messageClass, (returnDefault ? Schema.BYTES : null));
 	}
 
-	@SuppressWarnings({ "unchecked" })
+	@Override
+	@SuppressWarnings("unchecked")
+	public <T> Schema<T> getSchema(SchemaType schemaType, @Nullable ResolvableType messageType) {
```

Comment:
@sobychacko you can use this API in the binder and pass in null for `messsageType` which will give mappings for all of the primitives. 

```java
	public Schema<?> toSchema(SchemaType schemaType) {
		SchemaResolver schemaResolver = new DefaultSchemaResolver();
		return schemaResolver.getSchema(schemaType, null);
	}
```

Then just need to make it possible for users to wire in custom mappings.